### PR TITLE
backend: add weather tool

### PR DESCRIPTION
**Pull request by denrou**
New Community Weather Service

Uses Langchain weather document loader to retrieve weather information about some place.

### CODE REVIEW

**Comment by denrou** [comment]
Sorry, I did not have time to test the modification, I still have some problem making cohere-toolkit work correctly on my computer right now (using colima and not docker). But the hackaton ends soon and wanted to make a contribution ;)

File: src/community/tests/tools/retrieval/test_weather.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from community.tools.retrieval.weather import WeatherDataLoader
+
+
+def test_pub_med_retriever():
```
Comment (tianjing-li) [inline src/community/tests/tools/retrieval/test_weather.py:4]: Should this test name be changed?

**Comment by tianjing-li** [comment]
Hey @denrou do you have time to fix up this PR? There's been significant changes since then

**Comment by tianjing-li** [comment]
@denrou Will be closing this for now

**Pull request closed without merging**